dev/bots/analyze.dart

```diff
@@ -25,8 +25,9 @@ final String flutterRoot = path.dirname(path.dirname(path.dirname(path.fromUri(P
 final String flutter = path.join(flutterRoot, 'bin', Platform.isWindows ? 'flutter.bat' : 'flutter');
 final String flutterPackages = path.join(flutterRoot, 'packages');
 final String flutterExamples = path.join(flutterRoot, 'examples');
-final String dart = path.join(flutterRoot, 'bin', 'cache', 'dart-sdk', 'bin', Platform.isWindows ? 'dart.exe' : 'dart');
-final String pub = path.join(flutterRoot, 'bin', 'cache', 'dart-sdk', 'bin', Platform.isWindows ? 'pub.bat' : 'pub');
+late final String dartSdk;
+final String dart = path.join(dartSdk, 'bin', Platform.isWindows ? 'dart.exe' : 'dart');
```

Comment:
isn't this guaranteed to fail?

Reply:
(if it's not guaranteed to fail it's very confusing code)

Reply:
All static variables with initializers are lazily evaluated, so this works fine provided that `main` sets `dartSdk` before any attempt is made to evaluate `dart` or `pub`.

What would help to clear up your confusion?  Would you like me to mark `dart` and `pub` as `late final`?  Restructure the code in some way?

Reply:
Yeah that's much less confusing, thanks.